core/capabilities/remote/executable/parallel_execution_limiter.go

```diff
@@ -0,0 +1,53 @@
+package executable
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"golang.org/x/sync/semaphore"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+var ErrMaxParallelExecutionLimitReached = errors.New("parallel execution limit reached")
+
+type parallelExecutionLimiter struct {
+	maxParallelTasks int64
+	sem              *semaphore.Weighted
+	wg               sync.WaitGroup
+	stopChan         services.StopChan
+}
+
+func newParallelExecutionLimiter(maxParallelTasks int64) *parallelExecutionLimiter {
+	return &parallelExecutionLimiter{
+		maxParallelTasks: maxParallelTasks,
+		sem:              semaphore.NewWeighted(maxParallelTasks),
+		stopChan:         make(services.StopChan),
+	}
+}
+
+// ExecuteTask executes a task in parallel, returning ErrMaxParallelExecutionLimitReached if the maximum number of
+// parallel executing tasks has been reached.  The task is executed with the given context.
+func (t *parallelExecutionLimiter) ExecuteTask(ctx context.Context, task func(ctx context.Context)) error {
+	if !t.sem.TryAcquire(1) {
+		return ErrMaxParallelExecutionLimitReached
```

Comment:
What's the point of failing? What are we going to do if this starts failing in production?

Reply:
indeed, I'm not really convinced it helps to introduce a limit.

Reply:
IMO we should introduce a limit, but rather than erroring if we've hit the limit we should wait to acquire the limit until the context is cancelled.

100 feels a bit low for the limit btw, I think it could be an order of magnitude or two larger.

Reply:
updated the code to limit and wait on context if no slots, upped the default to a 1000